var/spack/repos/builtin/packages/python/package.py

```diff
@@ -72,40 +73,41 @@ class Python(Package):
     depends_on("tk",  when="+tk")
     depends_on("tcl", when="+tk")
 
+    @when('@2.7,3.4:')
+    def patch(self):
+        # NOTE: Python's default installation procedure makes it possible for a
+        # user's local configurations to change the Spack installation.  In
+        # order to prevent this behavior for a full installation, we must
+        # modify the installation script so that it ignores user files.
+        ff = FileFilter('Makefile.pre.in')
+        ff.filter(
+            r'^(.*)setup\.py(.*)((build)|(install))(.*)$',
+            r'\1setup.py\2 --no-user-cfg \3\6'
+        )
+
     def install(self, spec, prefix):
+        # TODO: The '--no-user-cfg' option for Python installation is only in
+        # Python v2.7 and v3.4+ (see https://bugs.python.org/issue1180) and
+        # adding support for ignoring user configuration will require
+        # significant changes to this package for other Python versions.
+        if not spec.satisfies('@2.7,3.4:'):
+            tty.warn(('Python v{0} may not install properly if Python '
+                      'user configurations are present.').format(self.version))
+
         # Need this to allow python build to find the Python installation.
-        env['PYTHONHOME'] = prefix
+        env['PYTHONHOME'], env['PYTHONPATH'] = prefix, prefix
         env['MACOSX_DEPLOYMENT_TARGET'] = '10.6'
 
         # Rest of install is pretty standard except setup.py needs to
         # be able to read the CPPFLAGS and LDFLAGS as it scans for the
         # library and headers to build
-        include_dirs = [
-            spec['openssl'].prefix.include,  spec['bzip2'].prefix.include,
-            spec['readline'].prefix.include, spec['ncurses'].prefix.include,
-            spec['sqlite'].prefix.include,   spec['zlib'].prefix.include
-        ]
-
-        library_dirs = [
-            spec['openssl'].prefix.lib,  spec['bzip2'].prefix.lib,
-            spec['readline'].prefix.lib, spec['ncurses'].prefix.lib,
-            spec['sqlite'].prefix.lib,   spec['zlib'].prefix.lib
-        ]
-
-        if '+tk' in spec:
-            include_dirs.extend([
-                spec['tk'].prefix.include, spec['tcl'].prefix.include
-            ])
-            library_dirs.extend([
-                spec['tk'].prefix.lib, spec['tcl'].prefix.lib
-            ])
-
+        dep_pfxs = [dspec.prefix for dspec in spec.dependencies('link')]
```

Comment:
Wow, this is such a nicer way of generating a list of dependencies. I'll have to keep this in mind when I'm writing other packages.